docs/concepts/builder.rst

```diff
@@ -73,6 +73,22 @@ It encourages organizing code into logical modules (e.g., feature processing, mo
         .build()
     )
 
+.. note::
+
+    Your modules may have same named functions which will raise an error when using ``.build()`` since we cannot have two functions with the same name. You can use the method ``.allow_module_overrides()`` and Hamilton will choose the function from the later imported module.
```

Comment:
Two "nodes" with the same name -- fns -> nodes are 1:many.  Might be worth mentioning that.